qubes/storage/__init__.py

```diff
@@ -134,8 +134,12 @@ def __init__(self, name, pool, vid,
         self.source = source
         #: Volume unique (inside given pool) identifier
         self.vid = vid
+        #: Asynchronous lock for cryptsetup operations
+        self._crypt_lock = asyncio.Lock()
```

Comment:
What this lock guards against? Since it is per-volume lock, it won't prevent parallel cryptsetup calls for different volumes. And for a single volume, there is already VM-level lock that guarantee only a single start or stop is running at a time.